projects/pgai/pgai/vectorizer/vectorizer.py

```diff
@@ -942,7 +1236,13 @@ async def _fetch_work(self, conn: AsyncConnection) -> list[SourceRow]:
         """
         queue_table_oid = await self._get_queue_table_oid(conn)
         async with conn.cursor(row_factory=dict_row) as cursor:
-            if self.features.loading_retries:
+            if self.features.split_processing:
+                query = self.queries.fetch_work_query_with_split_transaction
+                await cursor.execute(
+                    query,
+                    (self._batch_size,),
+                )
+            elif self.features.loading_retries:
```

Comment:
the double backwards compatibility makes my brain hurt, but I guess as long as we don't have a way to force upgrade there is no way to simplify this.